cli/cfncluster/cfncluster.py

```diff
@@ -125,58 +131,62 @@ def update(args):
         master_subnet_id = config.parameters[i][1]
         try:
-            vpcconn = boto.vpc.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                 aws_secret_access_key=config.aws_secret_access_key)
-            availability_zone = str(vpcconn.get_all_subnets(subnet_ids=master_subnet_id)[0].availability_zone)
-        except boto.exception.BotoServerError as e:
-            logger.critical(e.message)
+            ec2 = boto3.client('ec2', region_name=config.region,
+                               aws_access_key_id=config.aws_access_key_id,
+                               aws_secret_access_key=config.aws_secret_access_key)
+            availability_zone = ec2.describe_subnets(SubnetIds=[master_subnet_id]) \
+                .get('Subnets')[0] \
+                .get('AvailabilityZone')
+        except ClientError as e:
+            logger.critical(e)
             sys.exit(1)
         config.parameters.append(('AvailabilityZone', availability_zone))
     except ValueError:
         pass
 
     try:
         logger.debug((config.template_url, config.parameters))
-        stack = cfnconn.update_stack(stack_name,template_url=config.template_url,
-                                     parameters=config.parameters, capabilities=capabilities,
-                                     disable_rollback=args.norollback)
-        status = cfnconn.describe_stacks(stack)[0].stack_status
+
+        cfn_params = [{'ParameterKey': param[0], 'ParameterValue': str(param[1])} for param in config.parameters]
+        stack = cfn.update_stack(StackName=stack_name,TemplateURL=config.template_url,
+                                 Parameters=cfn_params, Capabilities=capabilities)
+        status = stack.get('Stacks')[0].get('StackStatus')
         if not args.nowait:
             while status == 'UPDATE_IN_PROGRESS':
-                status = cfnconn.describe_stacks(stack)[0].stack_status
-                events = cfnconn.describe_stack_events(stack)[0]
-                resource_status = ('Status: %s - %s' % (events.logical_resource_id, events.resource_status)).ljust(80)
+                status = cfn.describe_stacks(StackName=stack_name).get("Stacks")[0].get('StackStatus')
+                events = cfn.describe_stack_events(StackName=stack_name).get('StackEvents')[0]
+                resource_status = ('Status: %s - %s' % (events.get('LogicalResourceId'), events.get('ResourceStatus'))).ljust(80)
                 sys.stdout.write('\r%s' % resource_status)
                 sys.stdout.flush()
                 time.sleep(5)
         else:
-            status = cfnconn.describe_stacks(stack)[0].stack_status
+            status = cfn.describe_stacks(StackName=stack_name).get("Stacks")[0].get('StackStatus')
             logger.info('Status: %s' % status)
-    except boto.exception.BotoServerError as e:
-        logger.critical(e.message)
+    except ClientError as e:
+        logger.critical(e)
```

Comment:
why not have `logger.critical(e.response.get('Error').get('Message'))`

Reply:
changed.